promql/functions.go

```diff
@@ -113,12 +113,19 @@ func extrapolatedRate(vals []parser.Value, args parser.Expressions, enh *EvalNod
 		}
 		// Handle counter resets:
 		prevValue := samples.Floats[0].F
+		resetCount := 0
 		for _, currPoint := range samples.Floats[1:] {
 			if currPoint.F < prevValue {
 				resultFloat += prevValue
+				resetCount++
 			}
 			prevValue = currPoint.F
 		}
+
+		resetRatio := float64(resetCount) / float64(numSamplesMinusOne)
+		if resetRatio > 0.6 {
```

Comment:
At least this should be a `const`, probably configurable.

Reply:
> probably configurable

the tricky part with a global configuration is that acceptable counter reset ratios might be metric specific?

Reply:
Noting that this is a ratio, please give an example of what you're thinking about.
Probably this discussion should be on the issue, which is where 0.6 is proposed.